pinot-spi/src/main/java/org/apache/pinot/spi/utils/JsonUtils.java

```diff
@@ -81,7 +81,7 @@ private JsonUtils() {
   public static final ObjectReader DEFAULT_READER = DEFAULT_MAPPER.reader();
   public static final ObjectWriter DEFAULT_WRITER = DEFAULT_MAPPER.writer();
   public static final ObjectWriter DEFAULT_PRETTY_WRITER = DEFAULT_MAPPER.writerWithDefaultPrettyPrinter();
-  private static final TypeReference<HashMap<String, Object>> GENERIC_JSON_TYPE =
+  public static final TypeReference<HashMap<String, Object>> GENERIC_JSON_TYPE =
```

Comment:
Suggest not exposing this one. We may add a helper method `stringToMap(String jsonString)`

Reply:
Why not? I understand we should not expose the ObjectMapper because it may be mutated, but object is immutable and may be useful to have a single instance if someone has to do some advance usage of Jackson.

I mean, I would think that JsonUtils is used to avoid repetition and useless object creation, not to prevent devs to know about Jackson. The same we expose DEFAULT_WRITER, DEFAULT_PRETTY_WRITER and DEFAULT_READER, we can (and I would say should) expose GENERIC_JSON_TYPE

Reply:
We can definitely expose common type reference. My only concern about exposing this type is that we name the `HashMap` type as `GENERIC_JSON` which is a little bit confusing to me, but I guess it might be okay

Reply:
I'm renaming it to `MAP_TYPE_REFERENCE`, which should be more clear.